p2p/simulations/adapters/types.go

```diff
@@ -265,3 +266,28 @@ func RegisterServices(services Services) {
 		os.Exit(0)
 	}
 }
+
+func (n *NodeConfig) initDefaultEnode() error {
+	return n.initEnode(net.IPv4(127, 0, 0, 1), 0, 0)
+}
+
+func (n *NodeConfig) initEnode(ip net.IP, tcpport int, udpport int) error {
+	enrIp := enr.IP(ip)
+	n.Record.Set(&enrIp)
+	enrTcpPort := enr.TCP(tcpport)
+	n.Record.Set(&enrTcpPort)
+	enrUdpPort := enr.UDP(tcpport)
+	n.Record.Set(&enrUdpPort)
+
+	err := enode.SignV4(&n.Record, n.PrivateKey)
+	if err != nil {
+		return fmt.Errorf("unable to generate ENR: %v", err)
+	}
+	nod, err := enode.New(enode.V4ID{}, &n.Record)
+	if err != nil {
+		return fmt.Errorf("unable to create enode: %v", err)
+	}
+	log.Trace("simnode new", "record", n.Record)
+	n.node = nod
```

Comment:
At first it feels very counter intuitive to me why a `NodeConfig` holds a `node` field. Because in simulations we exactly create a `Node` from the `NodeConfig`. Also, here we leave the ID empty. So intuitively it would be enough to store just the `enr.Recrod` what we already do. (So we also store redundantly.)

I guess my confusion is coming from the fact that the name 'node' is heavily overloaded. Though maybe I would still prefer the previous version what Felix did, that is creating the node on the fly.

```
func (n *NodeConfig) Node() *enode.Node {
	return enode.NewV4(&n.PrivateKey.PublicKey, net.IP{127, 0, 0, 1}, int(n.Port), int(n.Port))
}
```
commit: 30cd5c18549f645002aedb4c00e5bab683cb0835

Reply:
That node will not have the swarm ENR record. That's  the whole point of the change.

Reply:
`TypeOf(NodeConfig.Record) == TypeOf(enode.Node.r) // true` as I understand. So what I meant `NodeConfig.Record == NodeConfig.node.r` is redundantly stored.

Reply:
I find it inherent in the design of the sim package that Config transparently holds some of the node information that logically should be a direct member of `SimNode` or something. So then let's change that whole pattern.

In general I like instantiating an object once in memory, then just read it.

To create a new node here every time by adding the swarm records etc etc first is messy IMO.

Reply:
(I'm not going to change that whole pattern in this PR, though) :)